app/services/UIBuilder.scala

```diff
@@ -138,4 +139,42 @@ class UIBuilder {
       case x :: xs => partitionComponents(xs, errors, x +: others)
     }
 
+  private def fromSequenceWithLeadingCallout(sg: StackedGroup, calloutType: CalloutType, formData: Option[FormData])
+                              (implicit stanzaIdToUrlMap: Map[String, String]): Seq[UIComponent] = {
+
+    val (callouts, visualStanzas) = extractLeadingCallouts(calloutType, sg.group, Nil)
+
+    val groupSize: Int = sg.group.size
+
+    callouts.size match {
+      case 1 =>  fromStanzas(callouts.head +: stackStanzas(visualStanzas, Nil), Nil, formData)
+      case `groupSize` =>  Seq(fromYourCallGroup(callouts))
```

Comment:
these two lines can replaced by the single loading the fromStanzas() accumulator with the processed YourCall stanzas as you know you have >1 stacked callouts 
_ => fromStanzas(stackStanzas(visualStanzas, Nil), Seq(fromYourCallGroup(callouts)), formData)

Reply:
The call to "extractLeadingCallouts" has been replaced by two fairly simple lines. Also as the last two parts of the case statement can be replaced by a single line and the first case is no longer possible The case statement has been removed also.